datatransfer/impl/graphsync/graphsync_impl.go

```diff
@@ -102,20 +112,22 @@ type gsExtended interface {
 
 // getChannelIDAndData extracts extension data and creates a channel id then returns
 // both. Returns any errors.
-func (impl *graphsyncImpl) getChannelIDAndData(extendedData gsExtended) (datatransfer.ChannelID, *ExtensionDataTransferData, error) {
+func (impl *graphsyncImpl) getChannelIDAndData(extendedData gsExtended, initiator peer.ID) (datatransfer.ChannelID, error) {
```

Comment:
shouldn't this just be called getChannelID now?

Reply:
also I wonder if you just call this getTransferID and make it even simpler -- and then you can just construct the ChannelID's inline in the call to getChannel?